### PR TITLE
Feature/Fix: times_series limit for Druid

### PR DESCRIPTION
**Pull request by fabianmenges**
When setting s series limit for druid it currently returns the TopN for each individual datapoint instead of the the TopN for the entire time frame. See the before and after pictures attached.

This is probably more of a bug fix since the behavior after this change is consistent with the behavior that you see when you group by multiple columns.

Before:
before

After:
after

### CODE REVIEW

File: superset/connectors/druid/models.py [hunk 1]
```diff
@@ -798,6 +798,28 @@ def values_for_column(self,
     def get_query_str(self, query_obj, phase=1, client=None):
         return self.run_query(client=client, phase=phase, **query_obj)
 
+    def _add_filter_from_pre_query_data(self, df, dimensions, filter):
+        ret = filter
```
Comment (mistercrunch) [inline superset/connectors/druid/models.py]: `filter` is a reserved word which can be confusing, go for `filter_` or `filters`

**Review by mistercrunch — CHANGES_REQUESTED** [review]
Minor change requested. 

Sorry you had to dig through this piece of code, this is not the code I'm most proud of for sure... 

Otherwise looks good, though this large messy method is hard to review and hard to unit test, so I'll have to trust you on it and test against our staging.

Which version of Druid are you running? I was told by some of the Druid core committers that the issue you were seeing (around topN not showing fragments) has been fixed in more recent version, and was hoping that an upcoming upgrade would fix this, but we do see this on our side at the moment.

I'm also hoping that we eventually move Druid to use SQLAlchemy as well, but that'll require a proper SQLAlchemy dialect and that's assuming that Druid offers as much control/performance through the SQL option...

File: superset/connectors/druid/models.py (hunk 1 above)
Comment (fabianmenges) [reply superset/connectors/druid/models.py]: ah... keep forgetting filter :(

**Comment by fabianmenges** [comment]
We are running 0.10.1, we upgraded from 0.9.2 about 2 weeks ago and saw great performance gains across different query types. 

This changeset is to handle Druid behavior that is implemented like this by design (this is not addressing or working around a bug in Druid). If you specify a threshold of 5 for a TopN query, Druid will always return the top 5 results per granularity (lets ignore that its not guaranteed they are the actual top 5 but likely the top 5). 
As an example of the default TopN behavior, lets say you want to query the top 5 Campaigns with the most (ordered by) "Ad-Impression" on a daily level over a week. Druid will return the top 5 campaigns for Monday, the top 5 for Tuesday, etc... If everyday you have a different set of 5 campaigns you will end up with 7x5 = 35 different Campaigns in your result set, each one with exactly one datapoint. (This is what you can see in the first screenshot)

The argument for this changeset is, that the behavior described is not actually useful and counter intuitive when you want a line-chart for the top 5 Campaigns. What you expect is to see the change over time for the same top 5 Campaigns day by day over the course of the week.  
The way this is implemented is running one TopN query with granularity all over the entire time range to find the top 5 campaigns and then run a TopN query with granularity of 1 day filtered to the campaigns of the first run.

We have been running this code for about 3 weeks and have not run into problems.

**Pull request merged**